### ISSUE TITLE
Is the Session ID valid for API usage?

### ISSUE THREAD

**Issue by afawcett** [mshanemc/liveRecord#1]
Great demo video and solution! I wanted to ask though, did you find that the Session ID you got back works with the SF API's? I thought that was a no no?

**Comment by mshanemc** [comment]
So far, so good even on v40.  I haven't tried all the variations (using it from another component, or in another namespace).

Shane

On May 24, 2017, at 11:59 AM, Andrew Fawcett <notifications@github.com<mailto:notifications@github.com>> wrote:


Great demo video and solution! I wanted to ask though, did you find that the Session ID you got back works with the SF API's? I thought that was a no no?

**Comment by afawcett** [comment]
Indeed, it does, just tried it... however it is at odds with [this](https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/apex_api_calls.htm#apex_api_calls). I also see that my alternative impl [via  LCC](https://github.com/afawcett/streamingcomponent). Seems to have stopped working in my prerel... Hmmm..

**Comment by mshanemc** [comment]
LCC had been odd--they started using a new domain and I was hitting some DNS stuff they said might take a week or 2 to resolve.  Was going to go that same route originally for the same reason you did.

Shane

On May 28, 2017, at 2:40 PM, Andrew Fawcett <notifications@github.com<mailto:notifications@github.com>> wrote:


Indeed, it does, just tried it... however it is at odds with this<https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/apex_api_calls.htm#apex_api_calls>. I also see that my alternative impl via LCC<https://github.com/afawcett/streamingcomponent>. Seems to have stopped working in my prerel... Hmmm..

**Comment by afawcett** [comment]
Yeah it is a bit of a moving target, still its pre-release i guess. Despite supporting custom CSP, the latest release of Summer'17 has started adding connect-src 'self'. Which prevents WebSockets and thus Streaming API, thus rendering the LCC route currently useless. I've raised an issue here, https://github.com/developerforce/LightningContainerExamples/issues/4. I do hope we can get confirmation that one of these solutions will be supported by Salesforce. 👍
